dashboard/app/controllers/courses_controller.rb

```diff
@@ -100,4 +106,21 @@ def i18n_params
       :description_teacher
     ).to_h
   end
+
+  private
+
+  def student_redirect_course(course)
+    # Return nil if course is nil or we know the student can view the version requested.
+    return nil if !course || course.can_view_version?(current_user)
```

Comment:
wording nits: this method doesn't know anything about students, only `course.can_view_version?` does. so it seems like it would be better to either move the logic for detecting if the user is not a student into this method, or remove "student" from the method name and comment. (e.g. `def redirect_course`, and `# ... user can view`).

Reply:
yeah, great call--that method name doesn't make sense anymore. i'll rename to `redirect_course` and update the comment